Google.PowerShell/BigQuery/GcbqTable.cs

```diff
@@ -112,4 +112,145 @@ protected override void ProcessRecord()
+
+        /// <summary>
+        /// <para type="description">
+        /// User-friendly name for the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Description of the table.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        public string Description { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The lifetime of this table from the time of creation (in seconds).
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ByValues)]
+        [ValidateRange(3600, (long.MaxValue / 1000))]
+        public long Expiration { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            // Set up the Dataset based on parameters
+            Table newTab;
+            TablesResource.InsertRequest request;
+            switch (ParameterSetName)
+            {
+                case ParameterSetNames.ByObject:
+                    newTab = InputObject;
+                    request = Service.Tables.Insert(newTab, newTab.TableReference.ProjectId, 
+                        newTab.TableReference.DatasetId);
+                    break;
+                case ParameterSetNames.ByValues:
+                    newTab = new Table();
+                    newTab.TableReference = new TableReference();
+                    newTab.TableReference.DatasetId = Dataset;
+                    newTab.TableReference.ProjectId = Project;
+                    newTab.TableReference.TableId = TableId;
+                    newTab.FriendlyName = Name;
+                    newTab.Description = Description;
+                    if (Expiration != 0)
+                    {
+                        double currentMillis = DateTime.Now.ToUniversalTime().Subtract(
+                            new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc)
+                            ).TotalMilliseconds;
+                        newTab.ExpirationTime = (Expiration * 1000) + Convert.ToInt64(currentMillis);
+                    }
+                    request = Service.Tables.Insert(newTab, Project, Dataset);
+                    break;
+                default:
+                    throw UnknownParameterSetException;
+            }
+
+            // Add the new dataset to the project supplied.
+            try
+            {
+                Table response = request.Execute();
+                WriteObject(response);
+            }
+            catch (GoogleApiException ex) when (ex.HttpStatusCode == HttpStatusCode.Conflict)
+            {
+                WriteError(new ErrorRecord(ex,
+                    $"A table with the name {TableId} already exists in this dataset.",
```

Comment:
It would be great if you can add the name of the dataset here:
`$"A table with the name '{TableId}' already exists in the dataset '{Dataset}'."`